services/caldar-api/src/controllers/customer-controller.ts

```diff
@@ -0,0 +1,159 @@
+import fs from "fs";
+import path from "path";
+import { Request, RequestHandler, Response } from "express";
+import { Customer } from "@caldar/types";
+import { CustomerType } from "@caldar/types";
+
+const customersDataFilePath = path.join(
+  process.cwd(),
+  "src",
+  "data",
+  "customers.json"
+);
+
+type queryType = {
+  type?: CustomerType;
+};
+
+export const getCustomers: RequestHandler = (req: Request, res: Response) => {
+  let response = getDataCustomers();
+
+  res.status(200).json({
+    status: "Ok",
+    content: response,
+  });
+};
+
+export const getCustomersByType: RequestHandler = (
+  req: Request,
+  res: Response
+) => {
+  const query: queryType = {};
+
+  query.type = req.query.type ? Number(req.query.type) : undefined;
+
+  let customers = getDataCustomers();
+
+  let list = !query.type
+    ? customers
+    : customers.filter((x) => x.type === query.type);
+
+  res.status(200).json({
+    status: "Ok",
+    content: list,
+  });
+};
+
+export const createCustomer: RequestHandler = (req: Request, res: Response) => {
+  const newCustomer: Customer = req.body;
+
+  if (!newCustomer) {
+    res.status(400).json({
+      status: "error",
+      error: "New customer data must be present in the request body.",
+    });
+  }
+
+  let customers = getDataCustomers();
+
+  newCustomer.id = customers[customers.length - 1].id + 1;
+
+  const newList = [...customers, newCustomer];
+
+  saveData(newList);
+
+  res.status(200).json({
+    status: "ok",
+    item: newCustomer,
+  });
+};
+
+export const updateCustomer: RequestHandler = (req: Request, res: Response) => {
+  const updatedCustomer: Customer = req.body;
+
+  if (!updatedCustomer) {
+    res.status(400).json({
+      status: "error",
+      error: "Updated customer data must be present in the request body.",
+    });
+  }
+
+  if (!updatedCustomer.id) {
+    res.status(400).json({
+      status: "error",
+      error:
+        "The ID of the customer to update must be present in the request body.",
+    });
+  }
+
+  let customers = getDataCustomers();
+
+  let existCustomer = customers.some((x) => x.id === updatedCustomer.id);
+
+  if (!existCustomer) {
```

Comment:
Renamed variable for a better spelling
```suggestion
  if (!customerExists) {
```